test/decompiler/reference/engine/game/task/process-taskable_REF.gc

```diff
@@ -0,0 +1,2045 @@
+       (when (not (-> obj have-flava))
+        (set! (-> obj have-flava) #t)
+        (set-setting!
+         *setting-control*
+         pp
+         'sound-flava
+         #f
+         20.0
+         (the-as int (-> obj sound-flava))
+         )
+        )
+       )
+      ((-> obj have-flava)
+       (clear-pending-settings-from-process *setting-control* pp 'sound-flava)
+       (set! (-> obj have-flava) #f)
+       )
+      )
+     )
+    )
+   (when (-> obj music)
+    (let
+     ((s5-3
+       (vector-!
+        (new 'stack-no-clear 'vector)
+        (target-pos 0)
+        (the-as vector (-> obj root-override root-prim prim-core))
+        )
+       )
+      )
+     (set! (-> s5-3 y) (* 4.0 (-> s5-3 y)))
+     (cond
+      ((< (vector-length s5-3) 102400.0)
+       (when (not (-> obj have-music))
+        (set! (-> obj have-music) #t)
+        (set-setting!
+         *setting-control*
+         pp
+         'music
+         (the-as symbol (-> obj music))
+         0.0
+         0
+         )
+        )
+       )
+      ((-> obj have-music)
+       (clear-pending-settings-from-process *setting-control* pp 'music)
+       (set! (-> obj have-music) #f)
+       )
+      )
+     )
+    )
+   0
+   (none)
+   )
+  )
+
+;; definition for method 31 of type process-taskable
+;; INFO: Return type mismatch art-joint-anim vs art-element.
+(defmethod get-art-elem process-taskable ((obj process-taskable))
+  (the-as art-element (if (> (-> obj skel active-channels) 0)
+                       (-> obj skel root-channel 0 frame-group)
+                       )
+   )
+  )
+
+;; definition for method 32 of type process-taskable
+;; INFO: Return type mismatch symbol vs basic.
+(defmethod play-anim! process-taskable ((obj process-taskable) (arg0 symbol))
+  (the-as basic #f)
+  )
+
+;; definition for method 33 of type process-taskable
+;; INFO: Return type mismatch int vs none.
+(defmethod dummy-33 process-taskable ((obj process-taskable))
+  (let ((s5-0 (play-anim! obj #f)))
+   (if (type-type? (-> s5-0 type) spool-anim)
+    (spool-push *art-control* (-> (the-as spool-anim s5-0) name) 0 obj -99.0)
+    )
+   )
+  0
+  (none)
+  )
+
+;; definition for method 51 of type process-taskable
+(defmethod close-anim-file! process-taskable ((obj process-taskable))
+  (let* ((gp-0 (play-anim! obj #f))
+         (v1-2 (if (and (nonzero? gp-0) (type-type? (-> gp-0 type) spool-anim))
+                gp-0
+                )
+          )
+         )
+   (if v1-2
+    (file-status *art-control* (-> (the-as spool-anim v1-2) name) 0)
+    )
+   )
+  )
+
+;; definition for method 34 of type process-taskable
+(defmethod dummy-34 process-taskable ((obj process-taskable))
+  #f
+  )
+
+;; definition for method 35 of type process-taskable
+;; INFO: Return type mismatch int vs none.
+(defmethod dummy-35 process-taskable ((obj process-taskable))
+  (let* ((a0-1 obj)
+         (t9-0 (method-of-object a0-1 dummy-34))
+         )
+   (let ((s5-0 (the-as basic (t9-0 a0-1))))
```

Comment:
this dummy-34 method takes another argument - there's an `or a1, s7, r0` before the jalr.